File: day2/puzzle1.js
var fs = require('fs');

fs.readFile('day2.txt', 'utf-8', function (err, inp) {
 if (err) throw err;
 console.log(inp.trim().split(/\n/g).reduce(function (curr, line) {
  var m = line.match(/([0-9]+)x([0-9]+)x([0-9]+)/);
  var l = Number(m[1]), w = Number(m[2]), h = Number(m[3]);
  var area = (2*l*w)+(2*w*h)+(2*h*l), extra = Math.min(l*w,w*h,h*l);
  return curr + area + extra;
 }, 0));
});

File: 2015/02/puzzle1.py
import re

total = 0
for l in open('day2.txt'):
 m = re.match('([0-9]+)x([0-9]+)x([0-9]+)', l)
 l = int(m.group(1))
 w = int(m.group(2))
 h = int(m.group(3))
 area = (2*l*w)+(2*w*h)+(2*h*l)
 extra = min(l*w,w*h,h*l)
 total = total + area + extra
 print l, w, h, "=", area, extra
print "total", total
